_posts/2022-04-21-tardigrade-launch.md

```diff
@@ -0,0 +1,323 @@
+of different sizes (`10GB` / `1TB` / `10TB`). The queries were executed
+sequentially as well as with concurrency factors of 5, 10, and 20. Failure
+recovery capabilities were tested by crashing a random node in a cluster every
+couple of minutes while streaming a live workload.
+
+To validate new resource management capabilities we submitted all 22
+[TPC-H](https://github.com/trino/trino-verifier-queries/tree/main/src/main/resources/queries/tpch/etl)
+based queries simultaneously with fault tolerant execution on and off. With
+fault tolerant execution disabled only two of them succeeded while all the
+rest failed with resource related issues (running out of memory). With fault
+tolerant execution enabled all of them succeeded with no issues.
+
+## How Do I Enable Granular Retries?
+
+Fault tolerant execution can only be enabled for an entire cluster.
+
+In general, we recommend splitting your long-running ETL queries and
+short-running interactive queries into different clusters. This will ensure the
+long running ETL queries do not impact interactive workloads and cause a bad
+user experience. Please note that any short-running, interactive queries on a
+fault tolerant cluster may experience higher latencies due to the checkpoint
+mechanism.
+
+First you need to create an S3 bucket for spooling (currently only AWS S3 is
+supported). We recommend configuring a bucket lifecycle rule to
+automatically expire abandoned objects in an event of a node crash:
+<img src="/assets/blog/tardigrade-launch/bucket-lifecycle.png"/>
+
+Second you need to configure exchange manager. It can be done by creating the
+`etc/exchange-manager.properties` configuration file in the Trino working
+directory on coordinator and on each worker with the following configuration:
+  * `exchange-manager.name=filesystem`
+  * `exchange.base-directories=s3://<bucket-name>`
+  * `exchange.s3.region=us-east-1`
+  * `exchange.s3.aws-access-key=<access-key>`
+  * `exchange.s3.aws-secret-key=<secret-key>`
+
+Lastly, you need to configure and enable task level retries by changing
+`etc/config.properties`:
+  * `retry-policy=TASK`
+  * `query.hash-partition-count=50` (more than 50 is currently not supported by
+    the filesystem exchange implementation)
+
+It is also recommended to enable compression to reduce amount of data spooled
+on S3 (`exchange.compression-enabled=true`) as well as reduce the low memory
+killer delay to allow resource manager to unblock nodes running short on memory
+faster (`query.low-memory-killer.delay=0s`). Additionally, we recommend enabling
+automatic writer scaling to optimize output file size for tables created with
+Trino (`scale-writers=true`) which could be critical for using Trino as an ETL
+engine.
+
+To increase overall throughput and reduce resource related task retries we
+recommend adjusting the concurrency settings based on the hardware
+configuration you have chosen. Here is the settings for the hardware we've
```

Comment:
chosen.

Following are the settings for the hardware used in our testing (..)

and then again  .. use code block